examples/nodejs/src/experimental-features/applicable-inputs.ts

```diff
@@ -0,0 +1,512 @@
+      return action.choice.for_choice.choice_owner;
+  }
+}
+
+export async function getApplicableActions(
+  restClient: RestClient,
+  contractId: ContractId,
+  environment?: Environment
+): Promise<ApplicableAction[]> {
+  const contractDetails = await restClient.getContractById(contractId);
+  const currentContract = contractDetails.currentContract
+    ? contractDetails.currentContract
+    : contractDetails.initialContract;
+  const oneDayFrom = (time: Timeout) => time + 24n * 60n * 60n * 1000n; // in milliseconds
+  const now = datetoTimeout(new Date());
+  const nextTimeout = getNextTimeout(currentContract, now) ?? oneDayFrom(now);
+  const timeInterval = { from: now, to: nextTimeout - 1n };
+  const env = environment ?? { timeInterval };
+  if (typeof contractDetails.state === "undefined") {
+    // TODO: Check, I believe this happens when a contract is in a closed state, but it would be nice
+    //       if the API returned something more explicit.
+    return [];
+  }
+  const initialReduce = reduceContractUntilQuiescent(
+    env,
+    contractDetails.state,
+    currentContract
+  );
+  if (initialReduce == "TEAmbiguousTimeIntervalError")
+    throw new Error("AmbiguousTimeIntervalError");
+  const applicableActions: ApplicableAction[] = initialReduce.reduced
+    ? [
+        {
+          type: "Advance",
+          policyId: contractDetails.roleTokenMintingPolicyId,
+          applyAction() {
+            return {
+              inputs: [],
+              environment: env,
+              reducedState: initialReduce.state,
+              reducedContract: initialReduce.continuation,
+              warnings: convertReduceWarning(initialReduce.warnings),
+              payments: initialReduce.payments,
+            };
+          },
+        },
+      ]
+    : [];
+  const cont = initialReduce.continuation;
+  if (cont === "close") return applicableActions;
+  if ("when" in cont) {
+    const applicableActionsFromCases = await Promise.all(
+      cont.when.map((cse) =>
+        getApplicableActionFromCase(
+          restClient,
+          env,
+          initialReduce.continuation,
+          initialReduce.state,
+          initialReduce.payments,
+          convertReduceWarning(initialReduce.warnings),
+          cse,
+          contractDetails.roleTokenMintingPolicyId
+        )
+      )
+    );
+    return applicableActions.concat(
+      toApplicableActions(
+        applicableActionsFromCases.reduce(
+          mergeApplicableActionAccumulator.concat,
+          mergeApplicableActionAccumulator.empty
+        )
+      )
+    );
+  }
+  return applicableActions;
+}
+
+export async function mkPartyFilter(wallet: WalletAPI) {
+  const address = await wallet.getUsedAddresses();
+  const tokens = await wallet.getTokens();
+  let tokenMap = new Map<PolicyId, RoleName[]>();
+  function getTokenMap() {
+    if (tokenMap.size === 0 && tokens.length > 0) {
+      tokens.forEach((token) => {
+        // Role tokens only have 1 element
```

Comment:
Not always I believe, 
`e.g` : for the raffle we have duplicated the role tokens in case of the winners lost their tokens...

Should we filter out the Ada ? Same issue as there : https://github.com/input-output-hk/marlowe-ts-sdk/issues/153

Reply:
probably, I'll do it as part of PLT-9054